builder-base/scripts/install_golang.sh

```diff
@@ -30,68 +30,102 @@ source $SCRIPT_ROOT/common_vars.sh
-    local -r majorversion=$(if [[ $(echo "$version"|awk -F'.' '{print NF}') -ge 3 ]]; then echo ${version%.*}; else echo ${version%-*}; fi)
-    mkdir -p ${GOPATH}/go${majorversion}/bin
-    
-    for binary in go gofmt; do
-        ln -s /root/sdk/go${version}/bin/${binary} ${GOPATH}/go${majorversion}/bin/${binary}
-    done
-    
-    if [ "$version" != "$majorversion" ]; then
-        ln -s ${GOPATH}/bin/go${version} ${GOPATH}/bin/go${majorversion}
-    fi
+  local -r version=$1
+
+  # Removing the last number as we only care about the major version of golang
+  local -r majorversion=$(if [[ $(echo "$version" | awk -F'.' '{print NF}') -ge 3 ]]; then echo ${version%.*}; else echo ${version%-*}; fi)
+  mkdir -p ${GOPATH}/go${majorversion}/bin
+
+  for binary in go gofmt; do
+    ln -s /root/sdk/go${version}/bin/${binary} ${GOPATH}/go${majorversion}/bin/${binary}
+  done
+
+  if [ "$version" != "$majorversion" ]; then
+    ln -s ${GOPATH}/bin/go${version} ${GOPATH}/bin/go${majorversion}
+  fi
 }
 
-function build::go::install(){
-    # Set up specific go version by using go get, additional versions apart from default can be installed by calling
-    # the function again with the specific parameter.
-    local version=${1%-*}
-    build::go::extract $version
-    build::go::symlink $version
+function build::go::install {
+  # Set up specific go version by using go get, additional versions apart from default can be installed by calling
+  # the function again with the specific parameter.
+  local version=${1%-*}
+  build::go::extract $version
+  build::go::symlink $version
 }
 
-function build::go::extract() {
-    local version=$1
+function build::go::extract {
+  local version=$1
 
-    if [ $TARGETARCH == 'amd64' ]; then
-        local arch='x86_64'
-    else
-        local arch='aarch64'
-    fi
+  mkdir -p /tmp/go-extracted
+  cd /tmp/go-extracted
+  tar -xzf /tmp/linux/$TARGETARCH/go$version.linux-$TARGETARCH.tar.gz
 
-    mkdir -p /tmp/go-extracted
-    for rpm in /tmp/$arch/golang-*.noarch.rpm /tmp/$arch/golang-*.$arch.rpm ; do $(cd /tmp/go-extracted && rpm2cpio $rpm | cpio -idm && rm -f $rpm); done
+  local -r golang_version=$(/tmp/go-extracted/go/bin/go version | grep -o "go[0-9].* " | xargs)
 
-    local -r golang_version=$(/tmp/go-extracted/usr/lib/golang/bin/go version | grep -o "go[0-9].* " | xargs)
+  mkdir -p /root/sdk/$golang_version
+  mv /tmp/go-extracted/go/* /root/sdk/$golang_version
 
-    mkdir -p /root/sdk/$golang_version
-    mv /tmp/go-extracted/usr/lib/golang/* /root/sdk/$golang_version
+  version=$(echo "$golang_version" | grep -o "[0-9].*")
+  ln -s /root/sdk/go${version}/bin/go ${GOPATH}/bin/$golang_version
 
-    for license_dir in "/usr/share/licenses/golang" "/usr/share/doc/golang-$VERSION"; do
-        if [ -d /tmp/go-extracted/$license_dir ]; then
-            mv /tmp/go-extracted/$license_dir/* /root/sdk/$golang_version
-        fi
-    done
+  # This is only for 1.21+ and is causing failures.
```

Comment:
id probably jsut remove this completely here since its not relevant to 1.21+ versions and its preserved down below for older versions.